primedev/mods/modweaponvars.cpp

```diff
@@ -0,0 +1,311 @@
+
+	if (result && IsWeapon<ScriptContext::SERVER>((void**)(unk_2 - 0x1410)))
+	{
+		SQObject* entInstance = g_pSquirrel<ScriptContext::SERVER>->__sq_createscriptinstance((void**)(unk_2 - 0x1410));
+		g_pSquirrel<ScriptContext::SERVER>->Call("CodeCallback_ApplyModWeaponVars", entInstance);
+	}
+
+	return result;
+}
+
+ADD_SQFUNC("void", ModWeaponVars_SetInt, "entity weapon, int weaponVar, int value", "", ScriptContext::SERVER | ScriptContext::CLIENT)
+{
+	void** ent = g_pSquirrel<context>->getentity<void*>(sqvm, 1);
+	if (!IsWeapon<context>(ent))
+	{
+		g_pSquirrel<context>->raiseerror(sqvm, "Entity is not a weapon");
+		return SQRESULT_ERROR;
+	}
+	int weaponVar = g_pSquirrel<context>->getinteger(sqvm, 2);
+	int value = g_pSquirrel<context>->getinteger(sqvm, 3);
+	if (weaponVar < 1 || weaponVar > WEAPON_VAR_COUNT) // weapon vars start at one and end at 725 inclusive
+	{
+		// invalid weapon var index
+		g_pSquirrel<context>->raiseerror(sqvm, "Invalid eWeaponVar!");
+		return SQRESULT_ERROR;
+	}
+
+	WeaponVarInfo* varInfo = &weaponVarArray<context>[weaponVar];
+	if (varInfo->type != WVT_INTEGER)
+	{
+		// invalid type used
+		g_pSquirrel<context>->raiseerror(sqvm, "eWeaponVar is not an integer!");
+		return SQRESULT_ERROR;
+	}
+
+	if (context == ScriptContext::SERVER)
+	{
+		CWeaponX* weapon = (CWeaponX*)ent;
+		*(int*)(&weapon->weaponVars[varInfo->offset]) = value;
+	}
+	else // if (context == ScriptContext::CLIENT)
+	{
+		C_WeaponX* weapon = (C_WeaponX*)ent;
+		*(int*)(&weapon->weaponVars[varInfo->offset]) = value;
+	}
+
+	return SQRESULT_NULL;
+}
+
+ADD_SQFUNC("void", ModWeaponVars_SetFloat, "entity weapon, int weaponVar, float value", "", ScriptContext::SERVER | ScriptContext::CLIENT)
+{
+	void** ent = g_pSquirrel<context>->getentity<void*>(sqvm, 1);
+	if (!IsWeapon<context>(ent))
+	{
+		g_pSquirrel<context>->raiseerror(sqvm, "Entity is not a weapon");
+		return SQRESULT_ERROR;
+	}
+	int weaponVar = g_pSquirrel<context>->getinteger(sqvm, 2);
+	float value = g_pSquirrel<context>->getfloat(sqvm, 3);
+	if (weaponVar < 1 || weaponVar > WEAPON_VAR_COUNT) // weapon vars start at one and end at 725 inclusive
+	{
+		// invalid weapon var index
+		g_pSquirrel<context>->raiseerror(sqvm, "Invalid eWeaponVar!");
+		return SQRESULT_ERROR;
+	}
+
+	WeaponVarInfo* varInfo = &weaponVarArray<context>[weaponVar];
+	if (varInfo->type != WVT_FLOAT32)
+	{
+		// invalid type used
+		g_pSquirrel<context>->raiseerror(sqvm, "eWeaponVar is not a float!");
+		return SQRESULT_ERROR;
+	}
+
+	if (context == ScriptContext::SERVER)
+	{
+		CWeaponX* weapon = (CWeaponX*)ent;
+		*(float*)(&weapon->weaponVars[varInfo->offset]) = value;
+	}
+	else // if (context == ScriptContext::CLIENT)
+	{
+		C_WeaponX* weapon = (C_WeaponX*)ent;
+		*(float*)(&weapon->weaponVars[varInfo->offset]) = value;
+	}
+
+	return SQRESULT_NULL;
+}
+
+ADD_SQFUNC("void", ModWeaponVars_GetType, "int eWeaponVar", "", ScriptContext::SERVER | ScriptContext::CLIENT)
```

Comment:
this sqfunction should be marked with a int return type probably.

Reply:
oop my mistake, will fix tomorrow